parsl/executors/high_throughput/mpi_prefix_composer.py

```diff
@@ -25,13 +25,19 @@ def validate_resource_spec(resource_spec: Dict[str, str]):
         is invalid (e.g, contains invalid keys)
     """
     user_keys = set(resource_spec.keys())
-    legal_keys = set(("RANKS_PER_NODE",
-                      "NUM_NODES",
-                      "LAUNCHER_OPTIONS",
+    legal_keys = set(("ranks_per_node",
+                      "num_nodes",
+                      "num_tasks",
+                      "launcher_options",
                       ))
     invalid_keys = user_keys - legal_keys
     if invalid_keys:
         raise InvalidResourceSpecification(invalid_keys)
+    if "num_nodes" in resource_spec:
+        if not resource_spec.get("num_tasks") and resource_spec.get("ranks_per_node"):
+            resource_spec["num_tasks"] = resource_spec["num_nodes"] * resource_spec["ranks_per_node"]
```

Comment:
so there's a conflict of some kind if a user also specifies num_tasks, right?  - which in this case is that the user's num_tasks will be (silently) ignored?

Reply:
(more generally, it's not clear without turning my brain on, and so won't be clear to users immediately, which of the 8 combinations of having/not having these parameters are "sensible" vs "not sensible")

Reply:
That logic is supposed to set `num_tasks` and `ranks_per_node` if one is not set but the other is. I'm only using `num_tasks` because it is the kwarg used by flux. I feel like it is better to either call it `ranks` or `tasks`.